packages/build/src/plugins/node_version.js

```diff
@@ -65,43 +38,19 @@ const addPluginNodeVersion = function ({
   return { ...pluginOptions, nodePath: execPath, nodeVersion: currentNodeVersion }
 }
 
-const nonUIPluginNodeVersion = function ({
-  pluginOptions,
-  currentNodeVersion,
-  userNodeVersion,
-  nodePath,
-  featureFlags,
-}) {
-  // We want to slowly move node.js <10 plugin executions to use our system node version
-  if (
-    featureFlags.buildbot_build_plugins_system_node_version &&
-    satisfies(userNodeVersion, `<${MINIMUM_NODE_VERSION_SUPPORTED}`)
-  ) {
+const nonUIPluginNodeVersion = function ({ pluginOptions, currentNodeVersion, userNodeVersion, nodePath }) {
+  if (satisfies(userNodeVersion, `<${MINIMUM_NODE_VERSION_SUPPORTED}`)) {
     return { ...pluginOptions, nodePath: execPath, nodeVersion: currentNodeVersion }
   }
   return { ...pluginOptions, nodePath, nodeVersion: userNodeVersion }
 }
 
-// Ensure Node.js version is recent enough to run this plugin
+// Ensure Node.js version is compatible with plugin's `engines.node`
 const checkNodeVersion = function ({
   nodeVersion,
   packageName,
   pluginPackageJson: { engines: { node: pluginNodeVersionRange } = {} } = {},
 }) {
-  // Build plugins are instrumented with a wrapper code which has a minimal
-  // allowed Node.js version.
-  // A mismatch can only happen in the conditions described below since:
-  //  - production builds use a pinned version, except for local plugins and
-  //    plugins added to `package.json`
-  //  - Netlify CLI has the same minimal allowed Node.js version as
-  //    `@netlify/build`, so if users can run Netlify CLI, everything is good
-  if (!satisfies(nodeVersion, coreNodeVersionRange)) {
-    throwUserError(`The Node.js version is ${nodeVersion} but it should be ${coreNodeVersionRange} when using build plugins either:
-  - from the same repository (as opposed to npm modules)
-  - or added to "package.json"`)
-  }
-
-  // Plugins can also set a minimal version using `engines.node`
```

Comment:
This code is no longer executed as the `coreNodeVersionRange` is `>=8.3.0` - https://github.com/netlify/build/blob/591a2d7228fb66e4133990065e242e5a70a05fcf/packages/build/package.json#L129. Meaning if a plugin is being executed with something lower than that it will be executed with our system node version. (same will happen once we change this range to `>=10.18.0`)